client/my-sites/plans/current-plan/jetpack-product-install/index.tsx

```diff
@@ -161,34 +215,57 @@ export class JetpackProductInstall extends Component {
 	 * - We haven't retried too many times (limit is `MAX_RETRIES`).
 	 * - We currently have recoverable errors.
 	 *
-	 * @return {Boolean} Whether to trigger a request to refetch installation status.
+	 * @return Whether to trigger a request to refetch installation status.
 	 */
-	shouldRefetchInstallationStatus() {
+	shouldRefetchInstallationStatus(): boolean {
 		return this.retries < MAX_RETRIES && this.installationHasRecoverableErrors();
 	}
 
 	/**
 	 * A helper to refresh the page, which essentially will restart the installation process.
 	 *
-	 * @return {void}
+	 * @return {undefined} Eslint requires this silly return tag. @TODO get rid of this.
 	 */
-	refreshPage = () => void window.location.reload();
+	refreshPage = (): void => void window.location.reload();
 
 	/**
 	 * Request the current installation status.
 	 * Could be triggered by a timeout as we're waiting for installation to finish,
 	 * or by a retry if we discover we have a recoverable error.
 	 */
-	requestInstallationStatus = () => {
-		this.props.requestJetpackProductInstallStatus( this.props.siteId );
+	requestInstallationStatus = (): void => {
+		const { siteId } = this.props;
+		if ( ! siteId ) {
+			return;
+		}
+
+		this.props.requestJetpackProductInstallStatus( siteId );
 
 		if ( this.shouldRefetchInstallationStatus() ) {
 			this.retries++;
 		}
 	};
 
+	shouldRequestKeys(): boolean {
+		const { siteId, pluginKeys, requestedInstalls } = this.props;
+
+		if ( ! siteId ) {
+			return false;
```

Comment:
For some reason, I've seen these events fire recently:

https://github.com/Automattic/wp-calypso/blob/ffc35db2991cdca61b9b2539c8b7e25068f0b29d/client/my-sites/plugins/jetpack-plugins-setup/index.jsx#L193-L196

Would this be good place to add tracking for this in the new mechanism?